### ISSUE TITLE
when i open karabiner i dont find the controller already conected to my mac

### ISSUE THREAD

**Issue by Abdelkrim-zaouidi** [pqrs-org/Karabiner-Elements#2571]
karabiner seem to not detect my dualshok 4 controller wich can easily connect to my mac, but when i search for it in target devices i cant find it

**Comment by johans3** [comment]
I get the same issue, (MacOS 10.14.6) my generic joypad is missing from the list of available devices, the log tab does not register any errors, nor changes, no matter how often i plug my joypad in or out. The same applies for the Event viewer, no input at all, i tried every button on my device.
After the installation of **Carabiner Elements** (freshly installed yesterday, i shall have the newest version) i also restarted my computer to be sure that Carabiner elements is fully operational, as it is a package not just an application.

My mac is able to see my joypad. (Under: Apple>About this Mac>System info>USB ),  (i changed the labels from German to English if you wonder why some labels look different than usual labels)
```
Generic   USB  Joystick  :

  Product-ID:	0x0006
  Producer-ID:	0x0079
  Version:	1.07
  Speed:	Bis zu 1,5 MBit/s
  Producer:	DragonRise Inc.  
  Location-ID:	0xfd430000 / 9
  Available amperage (mA):	500
  Required amperage (mA):	500
  Additional operating current (mA):	0
```

Just to try another way, i stumbled over [Enjoyable.app](https://yukkurigames.com/enjoyable/) - which is a very old software, and i was able map my buttons, but some keys did overlap each other, in addition, the mouse pointer always moved left or right on-screen, without any user interaction, rendering that software useless. I did not add any turbo mode nor adding any strange configurations, also because deep customizations are not supported. Pity. Anyway,

Maybe it would not be a bad idea to add some depreciated input methods or other input methods to carabiner elements. I remember Carabiner elements served me very well years ago, when it was yet called **KeyRemap4MacBook**. I dont dare to install that old version as it is really old compared to my Mojave Os.

**Comment by s0lidarity** [comment]
I'm having the same issue on Catalina 10.15.6 
I have a Mayflash F300 USB joystick that is detected by the OS, I can remap via enjoyable, but it's not detected by Karabiner.

**Comment by s0lidarity** [comment]
I've tried this in Catalina and Big Sur. In both environments, the operating system can detect my Mayflash F300 as a USB input. It never gets detected by Karabiner though, I don't see any errors in the log when I plug it in or unplug it. Karabiner can detect my bluetooth keyboard just fine though. 

Is there a way I can try adding the device manually to the JSON file that shows my devices?

**Comment by MuhammedZakir** [comment]
> I've tried this in Catalina and Big Sur. In both environments, the operating system can detect my Mayflash F300 as a USB input. It never gets detected by Karabiner though, I don't see any errors in the log when I plug it in or unplug it. Karabiner can detect my bluetooth keyboard just fine though.
> 
> Is there a way I can try adding the device manually to the JSON file that shows my devices?

I don't think that will work. Because `key code` of joystick button will probably be different. Besides, how do you get device identifiers of that joystick?
